src/server.rs

```diff
@@ -721,6 +721,31 @@ impl LanguageServer for LspServer {
         };
         Ok(Some(response))
     }
+    async fn references(
+        &self,
+        params: lsp::ReferenceParams,
+    ) -> Result<Option<Vec<lsp::Location>>> {
+        let key =
+            params.text_document_position.text_document.uri.clone();
+        let store = self.store.lock().unwrap();
+        let contents = match store.get(&key) {
+            Some(v) => v,
+            None => {
+                error!(
+                    "textDocument/references called on unknown file {}",
+                    key
+                );
+                return Err(lspower::jsonrpc::Error::invalid_params(
+                    format!("file not opened: {}", key),
+                ));
+            }
+        };
+        let pos = params.text_document_position.position;
```

Comment:
Do we need this ownership change? Or can we just do `find_node(walk::Node::Package(&pkg), params.text_document_position.position)` ?

Reply:
Good catch. It's a bit verbose, but no, we don't need it.